lib/buildreq.c

```diff
@@ -175,7 +175,7 @@ int rc_aaa_ctx_server(rc_handle * rh, RC_AAA_CTX ** ctx, SERVER * aaaserver,
 
 		result = rc_send_server_ctx(rh, ctx, &data, msg, type);
 
-		if ((result == OK_RC) || (result == CHALLENGE_RC)) {
```

Comment:
On a REJECT we currently move to the next server in the list (if any). With the proposed change we would always return on a REJECT and not try the next server. That's a change of behavior that many users will not want.

I would say that using rc_set_flags()  certainly makes the code more explicit, but also we should check that there is indeed EAP contents to the REJECT before returning. In other words, on a REJECT with EAP contents and if the flag RC_FLAG_AVPAIR_ON_REJECT is set, then return.